internal/service/privatelinkendpointserviceserverless/data_source_privatelink_endpoints_service_serverless.go

```diff
@@ -101,7 +95,7 @@ func dataSourceMongoDBAtlasPrivateLinkEndpointsServiceServerlessRead(ctx context
 	return nil
 }
 
-func flattenServerlessPrivateLinkEndpoints(privateLinks []matlas.ServerlessPrivateEndpointConnection) []map[string]any {
+func flattenServerlessPrivateLinkEndpoints(privateLinks []admin.ServerlessTenantEndpoint) []map[string]any {
 	var results []map[string]any
```

Comment:
I think it's more clear constructions like:

var results []map[string]any

if len(privateLinks) == 0 {
   return results
}
results = make([]map[string]any, len(privateLinks))


to be like this:

if len(privateLinks) == 0 {
  return nil
}

results := make([]map[string]any, len(privateLinks))